platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/maps/MapboxMap.java

```diff
@@ -909,6 +909,52 @@ private void setApiBaseUrl(@NonNull MapboxMapOptions options) {
    */
   @UiThread
   public void setStyleUrl(@NonNull String url) {
+    setStyleUrl(url, null);
+  }
+
+  /**
+   * <p>
+   * Loads a new map style from the specified URL and receive a callback when the style has finished loading.
+   * </p>
+   * {@code url} can take the following forms:
+   * <ul>
+   * <li>{@code Style.*}: load one of the bundled styles in {@link Style}.</li>
+   * <li>{@code mapbox://styles/<user>/<style>}:
+   * retrieves the style from a <a href="https://www.mapbox.com/account/">Mapbox account.</a>
+   * {@code user} is your username. {@code style} is the ID of your custom
+   * style created in <a href="https://www.mapbox.com/studio">Mapbox Studio</a>.</li>
+   * <li>{@code http://...} or {@code https://...}:
+   * retrieves the style over the Internet from any web server.</li>
+   * <li>{@code asset://...}:
+   * reads the style from the APK {@code assets/} directory.
+   * This is used to load a style bundled with your app.</li>
+   * <li>{@code null}: loads the default {@link Style#MAPBOX_STREETS} style.</li>
+   * </ul>
+   * <p>
+   * This method is asynchronous and will return immediately before the style finishes loading.
+   * If you wish to wait for the map to finish loading listen for the {@link MapView#DID_FINISH_LOADING_MAP} event.
+   * </p>
+   * If the style fails to load or an invalid style URL is set, the map view will become blank.
+   * An error message will be logged in the Android logcat and {@link MapView#DID_FAIL_LOADING_MAP} event will be
+   * sent.
+   *
+   * @param url      The URL of the map style
+   * @param callback Callback that is invoked when the style has loaded.
+   * @see Style
+   */
+  @UiThread
+  public void setStyleUrl(@NonNull final String url, @Nullable final OnStyleLoadedListener callback) {
+    if (callback != null) {
+      nativeMapView.addOnMapChangedListener(new MapView.OnMapChangedListener() {
+        @Override
+        public void onMapChanged(@MapView.MapChange int change) {
+          if (change == MapView.DID_FINISH_LOADING_STYLE) {
+            callback.onStyleLoaded(url);
+            nativeMapView.removeOnMapChangedListener(this);
+          }
+        }
+      });
+    }
     nativeMapView.setStyleUrl(url);
```

Comment:
Why not call this class's `setStyleUrl()` in order to gate the native call through fewer (i.e. a single) places?